interceptor_track_local.go

```diff
@@ -0,0 +1,46 @@
+// +build !js
+
+package webrtc
+
+import (
+	"github.com/pion/rtp"
+)
+
+type interceptorChainTrackLocalWrapper struct {
+	TrackLocal
+	interceptor Interceptor
+}
+
+type interceptorTrackLocalWriter struct {
+	TrackLocalWriter
+	writeRTP WriteRTP
+}
+
+func WrapTrackLocal(track TrackLocal, interceptor Interceptor) TrackLocal {
```

Comment:
Instead of doing `Wrap...` would it be possible to rewrite stuff to assume an `Interceptor` is in use everywhere? Helper functions are sometimes harder to follow like this. You have to step into things when debugging.

Reply:
Could do this for TrackLocal, because I could wrap the TrackLocalWriter interface with one that runs the interceptors, then calls the original method. I'm having trouble with RemoteTrack though:

The TrackLocalWriter is implemented by WriteStreamSRTP in srtp package.
Similarly, there could be a TrackRemoteReader interface that could be implemented by ReadStreamSRTP. But TrackRemote does NOT use the ReadRTP function from ReadStreamSRTP, instead it is reimplemented in TrackRemote. I think this is because of peeking mechanism for determining the payload type. Would it be ok to move the payload type determination logic to srtp/ReadStreamSRTP? That would solve this issue, and avoid having the ReadRTP logic duplcated in TrackRemote.